leaks_test.go

```diff
@@ -83,6 +84,57 @@ func TestVerifyNone(t *testing.T) {
 	bg.unblock()
 }
 
+func TestIgnoreCurrent(t *testing.T) {
+	t.Run("Should ignore current", func(t *testing.T) {
+		defer VerifyNone(t)
+		done := make(chan struct{})
+		go func() {
+			<-done
+		}()
+		VerifyNone(t, IgnoreCurrent())
+		close(done)
+	})
+
+	t.Run("Should detect new leaks", func(t *testing.T) {
+		done := make(chan struct{})
+		defer VerifyNone(t)
```

Comment:
the use of multiple defers here actually makes the test harder to follow, can we simplify it?

e.g.,
```go
defer VerifyNone(t)

// There are no leaks currently.
VerifyNone(t)

done := make(chan struct{})
go func() {
  <-done
}()

err := Find()
require.Error(t, err, "Expected to find background goroutine as leak)

opt := IgnoreCurrent()
VerifyNone(opt)

// A second goroutine started after IgnoreCurrent is a leak
done2 := go func() {
  <-done2
}()
err := Find(opt)
require.Error(t, err, "Expect second goroutine to be flagged as a leak")

close(done)
close(done2)
```

Reply:
Thanks, your steps adapted and added as test case.